packages/nextclade/src/analyze/find_relative_mutations.rs

```diff
@@ -0,0 +1,137 @@
+use crate::alphabet::nuc::Nuc;
+use crate::analyze::aa_del::AaDel;
+use crate::analyze::aa_sub::AaSub;
+use crate::analyze::find_private_aa_mutations::{find_private_aa_mutations, PrivateAaMutations};
+use crate::analyze::find_private_nuc_mutations::{find_private_nuc_mutations, PrivateNucMutations};
+use crate::analyze::letter_ranges::{CdsAaRange, NucRange};
+use crate::analyze::nuc_del::NucDelRange;
+use crate::analyze::nuc_sub::NucSub;
+use crate::analyze::virus_properties::VirusProperties;
+use crate::coord::range::{AaRefRange, NucRefGlobalRange};
+use crate::gene::gene_map::GeneMap;
+use crate::translate::translate_genes::Translation;
+use crate::tree::tree::{AuspiceGraph, AuspiceRefNode};
+use eyre::{eyre, Report, WrapErr};
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+#[derive(Clone, Serialize, Deserialize, schemars::JsonSchema, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct RelativeNucMutations {
+  ref_node: AuspiceRefNode,
+  muts: PrivateNucMutations,
+}
+
+pub fn find_relative_nuc_mutations(
+  graph: &AuspiceGraph,
+  clade: &str,
+  clade_node_attrs: &BTreeMap<String, String>,
+  substitutions: &[NucSub],
+  deletions: &[NucDelRange],
+  missing: &[NucRange],
+  alignment_range: &NucRefGlobalRange,
+  ref_seq: &[Nuc],
+  non_acgtns: &[NucRange],
+  virus_properties: &VirusProperties,
+) -> Result<Vec<RelativeNucMutations>, Report> {
+  let ref_nodes = filter_ref_nodes(graph, clade, clade_node_attrs);
+
+  ref_nodes
+    .iter()
+    .map(|&ref_node| -> Result<_, Report> {
+      let node = graph
+        .iter_nodes()
+        .find(|node| node.payload().name == ref_node.name)
+        .ok_or_else(|| eyre!("Unable to find reference node on the tree: '{}'", &ref_node.name))?;
+
+      let muts = find_private_nuc_mutations(
+        node.payload(),
+        substitutions,
+        deletions,
+        missing,
+        alignment_range,
+        ref_seq,
+        non_acgtns,
+        virus_properties,
+      );
+
+      Ok(RelativeNucMutations {
+        ref_node: ref_node.to_owned(),
+        muts,
+      })
+    })
```

Comment:
There is very little new logic, mostly bookkeeping. The `find_private_*_mutations()` for nucs and aa are reused as is. Only difference is that it runs multiple times for all requested nodes.